### ISSUE TITLE
CFE doesn't reject bounds that contain raw types with non-simple bounds

### ISSUE THREAD

**Issue by stefantsov** [dart-lang/sdk#33208]
CFE should emit a compile-time error in the described case as specified in [this section](https://dart.googlesource.com/sdk/+/master/docs/language/informal/instantiate-to-bound.md#static-analysis) of the feature spec.

Currently, the following program compiles without any warnings or errors, and it should be rejected by the compiler with an appropriate compile-time error message:

```dart
class A<X extends Comparable<X>> {}

class B<Y extends A> {}

main() {
  B b;
}
```
The following program, that also should be rejected because of the use of generic types with non-simple bounds, makes the compiler to crash:

```dart
class C<Z extends C> {}

main() {
  C c;
}
```

**Comment by stefantsov** [comment]
The issue is being worked on in [CL 41264](https://dart-review.googlesource.com/c/sdk/+/41264).

**Comment by askeksa-google** [comment]
Fixing this should also fix #31810, #32416 and #32417.

**Comment by dgrove** [comment]
With -dev.59, the first example:

cat ~/tmp/dart2stable/33208a.dart
class A<X extends Comparable<X>> {}

class B<Y extends A> {}

main() {
  B b;
}
dgrove@dgrove:~$ /usr/lib/google-dartlang/bin/dartanalyzer --preview-dart-2 ~/tmp/dart2stable/33208a.dart 
Analyzing /usr/local/google/home/dgrove/tmp/dart2stable/33208a.dart...
  error • Type parameter bound types must be instantiated at tmp/dart2stable/33208a.dart:3:19 • strong_mode_not_instantiated_bound
  hint • The value of the local variable 'b' isn't used at tmp/dart2stable/33208a.dart:6:5 • unused_local_variable
1 error and 1 hint found.

cat ~/tmp/dart2stable/33208b.dart
class C<Z extends C> {}

main() {
  C c;
}
dgrove@dgrove:~$ /usr/lib/google-dartlang/bin/dartanalyzer --preview-dart-2 ~/tmp/dart2stable/33208b.dart 
Analyzing /usr/local/google/home/dgrove/tmp/dart2stable/33208b.dart...
   error • Type parameter bound types must be instantiated at tmp/dart2stable/33208b.dart:1:19 • strong_mode_not_instantiated_bound
  hint • The value of the local variable 'c' isn't used at tmp/dart2stable/33208b.dart:4:5 • unused_local_variable
1 error and 1 hint found.
